### DISCUSSION TITLE
Option to turn off "#" ?

### DISCUSSION THREAD

**Discussion by azer** [minad/consult#1091] [Q&A]
Hey there,

Thanks for creating consult -- it's the most stable navigation library for Emacs right now. 

The only friction in my experience is the "#" getting in the way. It not only feels very counter-intuitive design choice, also breaks the behavior for most users. The description in the repo also doesn't really tell how it benefits the user and why is it on by default. I found many people complaining that they can't get any results and they also don't know why.

Here's an example, no results just for finding file in the project:

<img width="380" alt="Screenshot 2024-09-19 at 22 51 24" src="https://github.com/user-attachments/assets/4929af84-11a3-4ed4-8ce1-5723674c56b2">

And here's after deleting `hash`: 

<img width="715" alt="Screenshot 2024-09-19 at 22 51 39" src="https://github.com/user-attachments/assets/536b5497-239b-4a2c-b22e-bc821cbf94c9">

**Comment by azer** [comment]
Just found out my answer: 

```el
(setq consult-async-split-style nil)
```

I'm leaving the issue open just to share my feedback about the default. Feel free to close it! 

And thanks for the great work ✨

**Comment by minad** [comment]
Thanks.

> The only friction in my experience is the "#" getting in the way. It not only feels very counter-intuitive design choice, also breaks the behavior for most users. The description in the repo also doesn't really tell how it benefits the user and why is it on by default. 

Did you read the manual? It is okay if the default setting does not work for you. However the setting certainly doesn't "break behavior for most users". Please don't make such generalizing statements which are not likely given how widely Consult is installed. I claim that many users understand and use the special filtering functionality provided by `#`. If `#` doesn't work or gets in the way, it may hint at some configuration problem. It would be interesting to investigate the reason behind these problems.

> I found many people complaining that they can't get any results and they also don't know why.

I haven't seen such complaints or bug reports. In case there is an actual issue I need a minimal recipe, see https://github.com/minad/consult?tab=readme-ov-file#bug-reports.
